js/modules/k6/experimental/sse/sse.go

```diff
@@ -0,0 +1,540 @@
+func (c *Client) On(event string, handler goja.Value) {
+	if handler, ok := goja.AssertFunction(handler); ok {
+		c.eventHandlers[event] = append(c.eventHandlers[event], handler)
+	}
+}
+
+func (c *Client) handleEvent(event string, args ...goja.Value) {
+	if handlers, ok := c.eventHandlers[event]; ok {
+		for _, handler := range handlers {
+			if _, err := handler(goja.Undefined(), args...); err != nil {
+				common.Throw(c.rt, err)
+			}
+		}
+	}
+}
+
+// closeResponseBody cleanly closes the response body.
+// Returns an error if sending the response body cannot be closed.
+func (c *Client) closeResponseBody() error {
+	var err error
+
+	c.shutdownOnce.Do(func() {
+		err = c.resp.Body.Close()
+		if err != nil {
+			// Call the user-defined error handler
+			c.handleEvent("error", c.rt.ToValue(err))
+		}
+		close(c.done)
+	})
+
+	return err
+}
+
+func (c *Client) pushSSEMetrics(connStart, connEnd time.Time) func() {
+	connDuration := metrics.D(connEnd.Sub(connStart))
+
+	metrics.PushIfNotDone(c.ctx, c.samplesOutput, metrics.ConnectedSamples{
+		Samples: []metrics.Sample{
+			{
+				TimeSeries: metrics.TimeSeries{
+					Metric: c.builtinMetrics.HTTPReqSending,
+					Tags:   c.tagsAndMeta.Tags,
+				},
+				Time:     connStart,
+				Metadata: c.tagsAndMeta.Metadata,
+				Value:    connDuration,
+			},
+		},
+		Tags: c.tagsAndMeta.Tags,
+		Time: connStart,
+	})
+
+	return func() {
+		end := time.Now()
+		requestDuration := metrics.D(end.Sub(connStart))
+
+		metrics.PushIfNotDone(c.ctx, c.samplesOutput, metrics.ConnectedSamples{
+			Samples: []metrics.Sample{
+				{
+					TimeSeries: metrics.TimeSeries{
+						Metric: c.builtinMetrics.HTTPReqs,
+						Tags:   c.tagsAndMeta.Tags,
+					},
+					Time:     end,
+					Metadata: c.tagsAndMeta.Metadata,
+					Value:    1,
+				},
+				{
+					TimeSeries: metrics.TimeSeries{
+						Metric: c.builtinMetrics.HTTPReqSending,
+						Tags:   c.tagsAndMeta.Tags,
+					},
+					Time:     end,
+					Metadata: c.tagsAndMeta.Metadata,
+					Value:    connDuration,
+				},
+				{
+					TimeSeries: metrics.TimeSeries{
+						Metric: c.builtinMetrics.HTTPReqDuration,
+						Tags:   c.tagsAndMeta.Tags,
+					},
+					Time:     end,
+					Metadata: c.tagsAndMeta.Metadata,
+					Value:    requestDuration,
+				},
+			},
+			Tags: c.tagsAndMeta.Tags,
+			Time: end,
+		})
+	}
+}
+
+// Wraps SSE in a channel
+func (c *Client) readEvents(readChan chan Event, errorChan chan error, closeChan chan int) {
+	reader := bufio.NewReader(c.resp.Body)
+	ev := Event{}
+	var buf bytes.Buffer
+
+	sendEvent := false
+	for {
+		line, err := reader.ReadBytes('\n')
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				select {
+				case closeChan <- -1:
+					return
+				case <-c.done:
+					return
+				}
+			} else {
+				select {
+				case errorChan <- err:
+					return
+				case <-c.done:
+					return
+				}
+			}
+		}
+
+		switch {
+		case hasPrefix(line, ":"):
+			// Comment, do nothing
```

Comment:
If we want a general solution, I think we can't ignore the comment. The user will test the SSE stream, which includes the comment...

Reply:
I do think that if there is a library that already parses this - it might be better to use it and if not - maybe split this as a library ;)